docs/stratis.txt

```diff
@@ -78,9 +89,8 @@ OPTIONS
         is also the default.
 --key-desc::
         The key description of the key that should be used to encrypt the
-        created pool. The key description must correspond to a kernel key
-        that is managed by the user and has at least one link to a searchable
-	keyring (thread, process, or session keyrings).
+        created pool. The key description must correspond to a key
```

Comment:
Last sentence revision:

```The key may be assigned to the key description in the kernel keyring using the *key* command.```

Because it could be done via the old school keyctl approach just as well, or with other mechanisms that we might be contemplating.

Reply:
This is true that they could theoretically do this, but it's going to be inconvenient and even difficult for users to do it through keyctl as we add a prefix to the key that is not public. Beyond that, we can't do key length checking if the user uses keyctl. Over all, I think it's a bad idea to allow the old approach as we lose some of the guardrails we've put in place.

Reply:
> This is true that they could theoretically do this, but it's going to be inconvenient and even difficult for users to do it through keyctl as we add a prefix to the key that is not public. Beyond that, we can't do key length checking if the user uses keyctl. Over all, I think it's a bad idea to allow the old approach as we lose some of the guardrails we've put in place.

@jbaublitz: Is the prefix added to the key or to the key description?

Reply:
It's added to the key description so the system key description displayed by keyctl will differ from what's displayed and used by Stratis at the application level. Otherwise we would not be able to tell which keys are set by Stratis and which ones are managed by other users. If we did not set this, we would theoretically allow a user to unlink a key in the keyring that is being used for something other than Stratis which I think we want to avoid.